spec/src/main/doc/cdi-se.asciidoc

```diff
@@ -0,0 +1,51 @@
+
+[source,java]
+----
+public static void main(String... args) {
+    CDIProvider provider = CDI.getCDIProvider();
+    CDI<Object> cdi = container.initialize();
+    // retrieve a bean and do work with it
+    MyBean myBean = cdi.select(MyBean.class).get();
+    myBean.doWork();
+    // when done
+    cdi.shutdown();
+}
+----
+
+The `CDI` utility class provides static lookup for `CDIProvider` instances based on `ServiceLoader` (this doesn't preclude the notion of manually instantiating the provider in a non-portable manner).  By default, the provider will be looked up via a ServiceLoader but can be over-written by calling `setCDIProvider` on the utility.  Once a `CDIProvider` has been located, it should not be instantiated again, instead the same one returned again.
+
+The `CDIProvider` instance returned is capable of starting a `CDI` instance.  By invoking either of the  `initialize` methods it can start up a container to retriever beans, fire events, gain access to the `BeanManager`.  The `initialize` methods on the `CDIProvider` interface must not be invoked when in an EE container.  If they are invoked, an `UnsupportedOperationException` may be thrown.  Likewise, if `shutdown` or `close` were invoked on the `CDI` class while in an EE container, an `UnsupportedOperationException` may also be thrown.
+
+Only a single `CDI` instance may be initialized within an SE application at a time, based on the provider used.  If a container has already been initialized and you call `initialize` again, an `IllegalStateException` is thrown.  If you manually instantiate a `CDIProvider` and initialize it, non-portable behavior may occur.
+
+The `CDI` can be stopped when it's not needed.  To help with this, `CDI` implements `AutoCloseable`, so when derefenced it should shutdown automatically (the `close` method invokes `shutdown`).  You may also shutdown the CDI instance manually by calling the `shutdown` method on the CDI object.  If the `shutdown` method is explicitly invoked, it may throw an `IllegalStateException` if the container was never initialized or if it was already shutdown.  The `close` method should not throw an `IllegalStateException` if the container was never initialized.
+
+Due to that, the following code block is equivalent to the above example:
+
+[source,java]
+----
+public static void main(String... args) {
+    try(CDI cdi = CDI.getCDIProvider().initialize()) {
+        // start the container, retrieve a bean and do work with it
+        MyBean myBean = cdi.select(MyBean.class).get();
+        myBean.doWork();
+    }
+    // shuts down automatically after the try with resources block.
+}
+----
+
+=== Classpath Scanning
+
+All classpath entries in a Java SE application are considered to be at the same level.  All rules that apply to bean resolution based on `beans.xml` declaration continue to apply the same way in Java SE.
```

Comment:
@johnament I see, so `foo/*` becomes `foo/bar.jar;foo/baz.jar`. Still there are other use-cases (except for WebStart and extensions) where this will probably not work - e.g. weld and owb arquillian embedded containers. My only concern is that if we stick with this solution from the beginning it might be harder to change/replace it later on.